### DISCUSSION TITLE
Server side search on read the docs

### DISCUSSION THREAD

**Discussion by Markeli** [squidfunk/mkdocs-material#2799] [Q&A]
Hi there!

My site built with mkdocs is hosted on readthedocs.org. ReadTheDocs provides server-side search with ElasticSearch. 
I've read docs about search customization, I realize that I can add server-side search support by myself. But I wonder that there is no embedded support for this search. 

Is there any article about configuring SSS on ReadTheDocs for a material theme?

**Comment by squidfunk** [comment]
I think there are some discussions with pointers on what needs to be done to embed third-party search, but in general, the search customization guide in the documentation should be the way to go.

**Comment by humitos** [comment]
Hi 👋🏼 . I work at Read the Docs and I found this issue while surfing the web 😄 

I wanted to chime in to note we are working in a completely new integration of the Read the Docs' features (including [Server Side Search](https://docs.readthedocs.io/en/latest/server-side-search/)) that's agnostic of the doctool. It works on Sphinx, MkDocs, Hugo, Docusaurus and many others. It's currently available in beta for projects using [`build.commands` config key](https://docs.readthedocs.io/en/latest/config-file/v2.html#build-commands). You can create a PR on your repository and give it a try by enabling [pull request previews](https://docs.readthedocs.io/en/latest/pull-requests.html) on you project.

Also, last week we published a blog post with more information about the work we are doing on these addons: https://blog.readthedocs.com/addons-flyout-menu-beta/

Feel free to open an issue at https://github.com/readthedocs/addons to report any problem or to share feedback. I don't have too much experience with Material for MkDocs, so if you had to do extra steps to setup your project on Read the Docs, it would be great that you share the steps/changes here as well 👍🏼
